pkg/apis/core/v1/validation/validation.go

```diff
@@ -90,6 +90,20 @@ func ValidateResourceQuantityValue(resource string, value resource.Quantity, fld
 			allErrs = append(allErrs, field.Invalid(fldPath, value, isNotIntegerErrorMsg))
 		}
 	}
+
+	// The quantity unit can only be one of E, P, T, G, M, K.
+	// Or use the power-of-two equivalents: Ei, Pi, Ti, Gi, Mi, Ki.
+	// e.g. "100m" is not allowed for memory, which will be parsed as milli.
+	if resource == string(v1.ResourceMemory) {
+		valueStr := value.String()
+		lastChar := valueStr[len(valueStr)-1:]
+		if lastChar != "i" {
```

Comment:
Tightening API validation is a breaking change that can cause problems with already persisted data. This seems like it will need to be caught/surfaced at runtime

Reply:
> Tightening API validation is a breaking change that can cause problems with already persisted data.

Right. It may involve incompatibility.

> caught/surfaced at runtime

Seems to be a better choice.

Reply:
> And pod update on resource is forbid as well. This means we can only delete the pod first and recreate it.

that's correct. the validation change in this PR means you will not be able to delete pods that are already in etcd with this value.